cpu/stm32f4/periph/spi.c

```diff
@@ -102,6 +111,9 @@ int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
             return -2;
     }
 
+    /* initialize lock */
+    mutex_init(&spi_config[dev].lock);
```

Comment:
OT: The problem with multiple calls to spi_init is that they can mess with each others' settings, e.g. baud rate. If there are two devices on the bus with different maximum speeds the code needs to be configured for the lowest baud rate of the two unless we want to manually switch rate between each SPI transfer.